src/main/java/de/tum/cit/aet/artemis/hyperion/service/AiQuizGenerationService.java

```diff
@@ -0,0 +1,152 @@
+    }
+
+    /**
+     * Generates an AI-based quiz for a given course using the specified generation parameters.
+     *
+     * @param courseId         the ID of the course for which the quiz should be generated
+     * @param generationParams the parameters that define how the quiz should be generated
+     * @return an {@link AiQuizGenerationResponseDTO} containing the generated quiz data
+     */
+    public AiQuizGenerationResponseDTO generate(long courseId, AiQuizGenerationRequestDTO generationParams) {
+        log.debug("Generating quiz for course {}", courseId);
+
+        try {
+            // load prompts from external templates
+            String systemPrompt = templateService.render("/prompts/hyperion/quiz_generation_system.st", Map.of());
+            String userPrompt = templateService.render("/prompts/hyperion/quiz_generation_user.st", generationParams.toTemplateVariables());
+
+            // Use Spring AI's structured output for better parsing
+            List<GeneratedMcQuestionDTO> questions = chatClient.prompt().system(systemPrompt).user(userPrompt).call().entity(new ParameterizedTypeReference<>() {
+            });
+
+            // Validate all questions
+            validateQuestions(questions);
+
+            log.info("Successfully generated {} quiz questions for course {}", questions.size(), courseId);
+            return new AiQuizGenerationResponseDTO(questions, List.of());
+        }
+        catch (IllegalArgumentException e) {
+            log.error("Validation failed for generated questions in course {}: {}", courseId, e.getMessage(), e);
+            return new AiQuizGenerationResponseDTO(List.of(), List.of("Generated questions failed validation: " + e.getMessage()));
+        }
+        catch (Exception e) {
+            log.error("Unexpected error during quiz generation for course {}: {}", courseId, e.getMessage(), e);
+            return new AiQuizGenerationResponseDTO(List.of(), List.of("Error during quiz generation: " + e.getMessage()));
+        }
+    }
+
+    /**
+     * Validates all generated questions according to business rules.
+     * Throws IllegalArgumentException if any question fails validation.
+     */
+    private void validateQuestions(List<GeneratedMcQuestionDTO> questions) {
+        if (questions == null || questions.isEmpty()) {
+            throw new IllegalArgumentException("No questions were generated");
+        }
+
+        for (int i = 0; i < questions.size(); i++) {
+            GeneratedMcQuestionDTO question = questions.get(i);
+            try {
+                validateQuestion(question);
+            }
+            catch (IllegalArgumentException e) {
+                throw new IllegalArgumentException("Question " + (i + 1) + ": " + e.getMessage(), e);
+            }
+        }
+    }
+
+    /**
+     * Validates a single question according to business rules.
+     */
+    private void validateQuestion(GeneratedMcQuestionDTO question) {
```

Comment:
Since you already annotated the DTOs with these validations, you could probably use them for automated validation. This would remove some of the manual handling here.